File: url-router/src/main/java/com/bukalapak/urlrouter/Result.java
package com.bukalapak.urlrouter;

import android.os.Bundle;

/**
 * Created by mrhabibi on 5/26/17.
 */

public class Result {
    public final String url;
    public final CastMap variables;
    public final CastMap queries;
    public final Bundle args;

    Result(String url, CastMap variables, CastMap queries, Bundle args) {
        this.url = url;
        this.variables = variables;
        this.queries = queries;
        this.args = args;
    }
}

File: url-router/src/main/java/com/bukalapak/urlrouter/Result.kt
package com.bukalapak.urlrouter

import android.content.Context
import android.os.Bundle

/**
 * Created by mrhabibi on 5/26/17.
 */

class Result(val context: Context,
             val url: String,
             val variables: ParamMap,
             val queries: ParamsMap,
             val fragment: String?,
             val args: Bundle?)

class RawResult(val variables: ParamMap = ParamMap(),
                val queries: ParamsMap = ParamsMap(),
                var fragment: String? = null) {

    fun cook(context: Context,
             url: String,
             args: Bundle?): Result = Result(
            context = context,
            url = url,
            variables = variables,
            queries = queries,
            fragment = fragment,
            args = args
    )
}